package/scripts/prmon_plot.py

```diff
@@ -257,24 +312,29 @@ def main():
 
     # Here comes the figure and data extraction
     fig, ax1 = plt.subplots()
-    xdata = np.array(data[xlabel]) * xmultiplier
+
+    xdata = []
     ydlist = []
-    for carg in ylist:
-        if args.diff:
-            num = np.array(data[carg].diff()) * ymultiplier
-            denom = np.array(data[xlabel].diff()) * xmultiplier
-            ratio = np.where(denom != 0, num / denom, np.nan)
-            ydlist.append(ratio)
-        else:
-            ydlist.append(np.array(data[carg]) * ymultiplier)
+
+    for i in range(len(data)):
+        xdata.append(np.array(data[i][xlabel] * xmultiplier))
+        ydlist.append(make_list(ylist, data[i], args, xmultiplier, ymultiplier, xlabel))
+
+    # Plot the graphs
+    line_styles = list(mpl.lines.lineStyles.keys())
+
     if args.stacked:
-        ydata = np.vstack(ydlist)
-        plt.stackplot(
-            xdata, ydata, lw=2, labels=[LEGENDNAMES[val] for val in ylist], alpha=0.6
-        )
+        if len(inputs) == 1:
+            for i in range(len(xdata)):
+                draw_stacked_graph(xdata[i], ydlist[i], ylist)
+        else:
+            print("ERROR:: Stacked graphs are not supported for more than one data set")
+            sys.exit(-1)
     else:
-        for cidx, cdata in enumerate(ydlist):
-            plt.plot(xdata, cdata, lw=2, label=LEGENDNAMES[ylist[cidx]])
+        for i in range(len(xdata)):
+            draw_line_graph(xdata[i], ydlist[i], ylist, line_styles[i % 7], inputs, i)
```

Comment:
If we drop the magical `7` then this'll need to be updated.

Reply:
The reason the MOD 7 is used is because the list of line styles is only of length 7. Thus if we get more than 7 types of line we will get a index out of range.

Reply:
However, you're using `7` as a magic number - what if a future release of matplotlib has more or less line styles?

At L318 you have:

```py
    line_styles = list(mpl.lines.lineStyles.keys())
```

Which suggests that the correct code would be `line_styles[i % len(line_styles)]`, right?

Reply:
Yes, that's a good point. Making these changes will ensure compatibility with future updates. I have made the necessary changes!

Reply:
Thank you, looks good now!

Reply:
Ah no worries. I just reformatted it with black.